### ISSUE TITLE
Cannot file module error with "module-templates.handlebarsConfig"

### ISSUE THREAD

**Issue by iilyak** [asbjornh/vscode-module-templates#26]
`.vscode/settings.json`

```json
{
    "module-templates.engine": "handlebars",
    "module-templates.handlebarsConfig": "handlebars.config.js",
    "module-templates.templateFiles": [
        "./decission.module-templates.json"
    ]
}
```

`.vscode/handlebars.config.js`

```JavaScript
module.exports = {
  helpers: {
    myHelper: () => "Hello!",
  },
};
```

`.vscode/decission.module-templates.json`
```
{
    "decission": {
        "displayName": "New ADR",
        "files": [
            {
              "name": "test.md",
              "content": [
                "hello {{myHelper}}"
              ]
            }
        ]
    }
}
```
Error message

```
Error: Cannot find module '<full_path>/.vscode/handlebars.config.js' in Error: Cannot find module
```

**Comment by iilyak** [comment]
More info
```
Cannot find module '<full_path>/.vscode/handlebars.config.js'
	at ~/.vscode-oss/extensions/asbjornh.vscode-module-templates-1.9.2/dist/extension.js:1:13953
	at Object.e.getHbsConfig (~/.vscode-oss/extensions/asbjornh.vscode-module-templates-1.9.2/dist/extension.js:1:14021)
	at ~/.vscode-oss/extensions/asbjornh.vscode-module-templates-1.9.2/dist/extension.js:1:28309
	at Generator.next (<anonymous>)
	at s (~/.vscode-oss/extensions/asbjornh.vscode-module-templates-1.9.2/dist/extension.js:1:27316)
	at runMicrotasks (<anonymous>)
	at processTicksAndRejections (internal/process/task_queues.js:93:5)
[2021-12-24 11:13:16.714] [renderer4] [error] [Extension Host] Error: Error: Cannot find module '<full_path>/.vscode/handlebars.config.js' in Error: Cannot find module '<full_path>/.vscode/handlebars.config.js'
	at ~/.vscode-oss/extensions/asbjornh.vscode-module-templates-1.9.2/dist/extension.js:1:13953
	at Object.e.getHbsConfig (~/.vscode-oss/extensions/asbjornh.vscode-module-templates-1.9.2/dist/extension.js:1:14021)
	at ~/.vscode-oss/extensions/asbjornh.vscode-module-templates-1.9.2/dist/extension.js:1:28309
	at Generator.next (<anonymous>)
	at s (~/.vscode-oss/extensions/asbjornh.vscode-module-templates-1.9.2/dist/extension.js:1:27316)
	at runMicrotasks (<anonymous>)
	at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

**Comment by iilyak** [comment]
I tried to recompile the extension and I've got a warning
```
WARNING in ./node_modules/handlebars/lib/index.js 24:2-20
require.extensions is not supported by webpack. Use a loader instead.
 @ ./source/utils.ts 38:32-53
 @ ./source/extension.ts
```

**Comment by iilyak** [comment]
Looks like similar issue https://github.com/fernandoescolar/vscode-solution-explorer/commit/539686e808f685d4d095700f9fc71093724518d5

**Comment by iilyak** [comment]
Confirmed this fixes the issue (in an ugly way):
```
diff --git a/source/utils.ts b/source/utils.ts
index bb39f85..818cf1a 100644
--- a/source/utils.ts
+++ b/source/utils.ts
@@ -81,7 +81,7 @@ export function getEngine(root: WorkspaceFolder): Engine {

 const tryRequire = (path: string) => {
   try {
-    return require(path);
+    return eval(`require('${path}')`);
   } catch (error) {
     throw new Error(
       `${error.name}: ${error.message} in ${error.stack.split("\n")[0]}`,
```

**Comment by iilyak** [comment]
https://www.typescriptlang.org/docs/handbook/migrating-from-javascript.html#importing-from-modules
Maybe I just needed to do 

```
declare function require(path: string): any;
```

**Comment by iilyak** [comment]
In the [vscode-handlebars-preview](https://github.com/johnknoop/vscode-handlebars-preview/blob/master/src/helpers.ts) extension the problem of loading helpers is solved differently. The extension watches the changes in the js file where helpers are defined. Therefore there is no need to reload vscode after each change.

**Comment by iilyak** [comment]
> https://www.typescriptlang.org/docs/handbook/migrating-from-javascript.html#importing-from-modules Maybe I just needed to do
> 
> ```
> declare function require(path: string): any;
> ```

This ^^^ didn't work

**Comment by asbjornh** [comment]
Thanks for reporting this issue!
I wasn't aware of if because this doesn't actually happen when debugging the extension inside of VSCode.

I've published version 1.9.3 that should fix it. Let me know if it helps

**Comment by iilyak** [comment]
Thank you for the fix. It works as expected now.

**Comment by martinfalke** [comment]
I still have the problem (Windows) and after some attempts of moving the file and adjusting the source code, I discovered that the issue is Windows paths using backslashes (`\`). This causes a bunch of characters to be escaped in the argument passed to `eval`, resulting in a path, with no distinction for sub-directories. I had to update a few of the dependencies for it to work, though. Namely `ts-loader`, `webpack`, and `webpack-cli`.

Compilation still warns about `require.extensions` not being supported by webpack, encouraging the use of a loader instead. Although it would probably make for a more elegant solution, I just wanted this to work locally and figured I could share so I took the shorter path to a working version.

I've forked the repo and will post a pull request soon. Thanks for the extension, it has served me well!

**Comment by asbjornh** [comment]
Is the file not found if you use normal forward slashes? Pretty sure `require` figures that out. Alternatively the backslashes probably need to be escaped?

**Comment by martinfalke** [comment]
I had initially placed the config file under the same directory as the extension `settings.json` file (`.vscode/`). In other words, all of the path parsing was performed by the extension code. I.e. there were no slashes in the config property.

I also tried to modify the extension's require-line to replace backslashes with double back-slashes (which probably would have worked had I replaced all, not just one), but I found it more elegant to simply switch to forward slashes.

Additionally, I tried placing the config file in a sub-directory, referencing it both with forward slashes and escaped backslashes. Neither worked, unfortunately.
